packages/mattermost-redux/src/actions/threads.ts

```diff
@@ -268,20 +260,46 @@ export function markAllThreadsInTeamRead(userId: string, teamId: string) {
     };
 }
 
-export function markThreadAsUnread(userId: string, teamId: string, threadId: string, postId: string) {
+export function markThreadAsUnread(userId: string, teamId: string, threadId: string) {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
-        try {
-            await Client4.markThreadAsUnreadForUser(userId, teamId, threadId, postId);
-        } catch (error) {
-            forceLogoutIfNecessary(error, dispatch, getState);
-            dispatch(logError(error));
-            return {error};
+        const getPostsForThread = makeGetPostsForThread();
+        let posts = getPostsForThread(getState(), threadId);
+
+        // load posts in thread if they are not loaded already
+        if (posts.length < 2) {
+            getPostThread(threadId)(dispatch, getState).then(({data, error}) => {
+                if (data) {
+                    posts = getPostsForThread(getState(), threadId);
+                    markThreadAsUnreadForUser(dispatch, getState, {userId, teamId, threadId, lastPostId: posts[0].id});
+                } else if (error) {
+                    return {error};
+                }
+                return {};
+            });
+        } else {
+            markThreadAsUnreadForUser(dispatch, getState, {userId, teamId, threadId, lastPostId: posts[0].id});
```

Comment:
We better add a new function e.g. `markLastPostInThreadAsUnread` and don't change this one, since we use it to mark replies as unread at will.
I.e. you can mark a reply as unread in a thread through that reply's dot menu (or by hitting alt(option)+Enter.

Reply:
Good point that I wasn't aware. Now new `markLastPostInThreadAsUnread` function is created.